src/main/java/calculator/calculator/StringCalculator.java

```diff
@@ -0,0 +1,39 @@
+package calculator.calculator;
+
+import calculator.global.ErrorMessage;
+import java.math.BigInteger;
+import java.util.List;
+import java.util.Objects;
+
+public class StringCalculator implements Calculator {
+
+    private final TokenProvider tokenProvider;
+
+    public StringCalculator(TokenProvider tokenProvider) {
+        this.tokenProvider = tokenProvider;
+    }
+
+    @Override
+    public BigInteger calculate(String rawInput) {
+        validateInput(rawInput);
+        List<String> tokens = tokenProvider.getTokens(rawInput);
+
+        if (isAllBlank(tokens)) {
+            return BigInteger.ZERO;
+        }
+
+        return tokens.stream()
+                .map(BigInteger::new)
+                .reduce(BigInteger.ZERO, BigInteger::add);
+    }
+
+    private void validateInput(String input) {
+        if (Objects.isNull(input)) {
+            throw new IllegalArgumentException(ErrorMessage.NULL_NOT_ALLOWED.getMessage());
+        }
+    }
+
+    private boolean isAllBlank(List<String> tokens) {
+        return tokens.stream().allMatch(String::isBlank);
+    }
```

Comment:
```suggestion
private boolean isAllBlank(List<String> tokens) {
        return tokens.stream()
                .allMatch(String::isBlank);
    }
```
메서드 체인을 줄바꿈으로 구분하면 더 가독성이 좋을 것 같아요~